apps/client/src/app/services/internet-identity.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthClient } from '@dfinity/auth-client';
+import { OAuthResponse } from '@ghostfolio/common/interfaces';
+import { EMPTY } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InternetIdentityService {
+  public constructor(private http: HttpClient) {}
+
+  public async login(): Promise<OAuthResponse> {
+    const authClient = await AuthClient.create();
+
+    return await new Promise((resolve, reject) => {
```

Comment:
not sure you need `await new`, maybe `new Promise...` is enough?

Reply:
Great catch! `return new Promise()` will do it.